src/App.tsx

```
@@ -1,37 +1,66 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type AppState = {
+  clockName: string;
+  hasClock: boolean;
+};
+
+export class App extends React.Component<{}, AppState> {
+  nameTimerId?: number;
+
+  contextHandler = (event: MouseEvent) => {
```

This method name violates the checklist naming convention. The checklist requires: "Name event handler methods to start with a verb (e.g., handleLeftClick, handleRightClick, handleContextMenu) rather than nouns (see checklist item 3)." Consider renaming `contextHandler` to a verb-based name such as `handleContextMenu` or `handleRightClick` to match the requirement and the project's style guide. Functionality (preventDefault + hiding) is correct, this is purely a naming/requirements issue.